In [1]:
import pandas as pd

In [2]:
df = pd.DataFrame() 

In [3]:
df = pd.read_csv("titanic_data_ESCP.csv")

df.head()

In [7]:
df.columns

Index(['PassengerId', 'Survived', 'Name', 'Sex', 'Age', 'Siblings - Spouse',
       'Parents - Children', 'Ticket', 'Ticket class', 'Fare', 'Cabin',
       'Port of embarkation'],
      dtype='object')

Q1: What is the total number of people on the titanic and how many of them survived and how many did not?

In [10]:
total_num = df['PassengerId'].max()

In [11]:
total_num

1309

In [13]:
survivors = len(df[df['Survived']==1])

In [14]:
survivors

342

Number of Passengers: 1309 while the number of survivors is 342

Q2: How many that survived were female and how many that died were female?

In [16]:
print(len(df[(df["Survived"]==0)&(df["Sex"]=="female")]))

81


The number of females that did not survive is 81

In [17]:
print(len(df[(df["Survived"]==1)&(df["Sex"]=="female")]))

233


The number of female survivors is 233

Q3: How many children were on the titanic? < 17.

In [18]:
len(df[df["Age"]<17])

134

Q4: How many children died that were on the ship?

In [20]:
len(df[(df["Survived"]==0) & (df["Age"]<17)])

45

45 children died

Q5: How many people had families with them?

In [22]:
len(df[df["Siblings - Spouse"]>0])

418

418 people

Q6: What is the ratio of female to male?

In [23]:
len(df[df["Sex"]=="female"]) / len(df[df["Sex"]=="male"])

0.5527876631079478

In [24]:
df.Sex.value_counts()

male      843
female    466
Name: Sex, dtype: int64